src/1_resources.js

```diff
@@ -27,14 +27,17 @@ var _validator;
 /**
  * @param {boolean} required
  * @param {validationTypes|RegExp} type
+ * @param {string=} fallback
+ * @param {validationTypes|RegExp=} fallbackType
  * @throws {Error}
- * @return {_validator}
  */
-function validator(required, type) {
-	return function(endpoint, param, data) {
+function validator(required, type, fallback, fallbackType) {
```

Comment:
hrm, i know this is a bit of a cop-out, but perhaps it would be cleaner to do this at the application layer (make app_id and branch_key optional, and in api.js, verify that at least one of them is present)?